### ISSUE TITLE
Support for Array items to be an Array (mixed types, i.e. tuple)

### ISSUE THREAD

**Issue by nandanrao** [atomicbits/scraml#75]
When attempting to parse a schema with this format: 

```
    "Tfidf": {
      "type": "array",
      "minItems": 2,
      "maxItems": 2,
      "items": [
        { "type": "number" },
        {
          "type": "array",
          "minItems": 2,
          "maxItems": 2,
          "items": [{ "type": "integer", "type": "number" }]
        }
      ]
    }
```

I receive the following error: 

```
io.atomicbits.scraml.jsonschemaparser.JsonSchemaParseException: An array type must have an 'items' field that refers to a JsObject in {"type":"array","minItems":2,"maxItems":2,"items":[{"type":"number"},{"type":"array","minItems":2,"maxItems":2,"items":[{"type":"number"}]}]}
```

JSON schema [array "items"] (http://json-schema.org/latest/json-schema-validation.html#anchor37) should support both objects and arrays. Support for arrays allows for a what is in essence a tuple, in javascript an array of mixed types. 

It is clear [here](https://github.com/atomicbits/scraml/blob/6591eadcc93e5c5043029f0a878410c202a344a7/modules/scraml-jsonschema-parser/src/main/scala/io/atomicbits/scraml/jsonschemaparser/model/ArrayEl.scala) that "items" in an array are only expected to be a JsObject, not a JsArray.

I would expect that "items" could be either a JsObject or a JsArray, and in the case of JsArray the schema supports arrays of mixed types. Would be happy to work on this if this is, indeed, a problem. Let me know - thanks!

**Comment by rigolepe** [comment]
Indeed, we don't support this json-schema feature yet. For now, arrays should only have a single type as items. I'm not sure yet how we should model this in Java and Scala. It seems like either an encapsulation object or an 'object' array would be possible here. Let's figure this out first before jumping to an implementation. 
We're focussing on RAML 1.0 support now, and there is quite some refactoring going on. It may be good to hold this one till we're through this.

**Comment by nandanrao** [comment]
Sounds like a plan!
